### ISSUE TITLE
Testing behavior not available in hk.Dropout

### ISSUE THREAD

**Issue by VIGNESHinZONE** [deepmind/dm-haiku#154]
Im still fairly new to `Haiku`, I was wondering why `hk.dropout` does not have any `is_training` parameter, as Dropout has different behavior during train and test time. 
Docs - https://dm-haiku.readthedocs.io/en/latest/api.html?highlight=Dropout#id2
Code - https://github.com/deepmind/dm-haiku/blob/b2ae68ecbec47e75f12eb318c40b0b9cb61b5bce/haiku/_src/basic.py#L299#L321

Whereas `hk.BatchNorm` has an `is_training` feature indicating the different behavior of batch_norm during training and testing. 
Docs - https://dm-haiku.readthedocs.io/en/latest/api.html?highlight=batchnorm#batchnorm

**Comment by tomhennigan** [comment]
Dropout is just a function, you can avoid calling it when you are doing inference:

```python
if is_training:
  x = hk.dropout(key, rate, x)
```

We have an example of this in our MLP net https://github.com/deepmind/dm-haiku/blob/master/haiku/_src/nets/mlp.py#L85#L118.

**Comment by VIGNESHinZONE** [comment]
thanks, @tomhennigan, I misunderstood the dropout implementation because usually during inference, the output layer is multiplied with `p_keep` to maintain the input distribution. But here in this implementation (also TensorFlow), `p_keep` is directly divided during training, so no additional operations are required during testing.

Reference [here](https://leimao.github.io/blog/Dropout-Explained/)
